dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerAccessManager.java

```diff
@@ -77,16 +77,35 @@ public class DefaultTrackerAccessManager implements TrackerAccessManager {
    */
   @Override
   public List<String> canRead(@Nonnull UserDetails user, TrackedEntity trackedEntity) {
-    // always allow if user == null (internal process) or user is superuser
     if (user.isSuper() || trackedEntity == null) {
```

Comment:
This block is not needed as its part of the `canRead` you created and call later

Reply:
I know, but I'd still leave it here.
The other method is just a temporary fix, and if remove this guard clause, it will be easy to forget to add it back when I remove the temporary method.